src/seedu/addressbook/data/person/Address.java

```diff
@@ -1,58 +1,145 @@
 package seedu.addressbook.data.person;
 
 import seedu.addressbook.data.exception.IllegalValueException;
+import java.util.*;
 
 /**
  * Represents a Person's address in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
-
-    public final String value;
-    private boolean isPrivate;
-
-    /**
-     * Validates given address.
-     *
-     * @throws IllegalValueException if given address string is invalid.
-     */
-    public Address(String address, boolean isPrivate) throws IllegalValueException {
-        this.isPrivate = isPrivate;
-        if (!isValidAddress(address)) {
-            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
-        }
-        this.value = address;
-    }
-
-    /**
-     * Returns true if a given string is a valid person email.
-     */
-    public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
-    }
-
-    @Override
-    public String toString() {
-        return value;
-    }
-
-    @Override
-    public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
-    }
-
-    @Override
-    public int hashCode() {
-        return value.hashCode();
-    }
-
-    public boolean isPrivate() {
-        return isPrivate;
-    }
+	public static final String EXAMPLE = "123, some street, some unit, some postal_code";
+	public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses should be in the format of BLOCK, STREET, UNIT, POSTAL CODE";
+	public static final String ADDRESS_VALIDATION_REGEX = ". ";
+	private static final int ADDRESS_PARAM_COUNT = 4;
+
+	public final String value;
+	private boolean isPrivate;
+	private Block block;
+	private Street street;
+	private Unit unit;
+	private PostalCode postalCode;
+
+	/**
+	 * Validates given address.
+	 *
```

Comment:
Good job on adding a feature to validate the addresses. 

However this comment header can be improved, as it is not clear from this method alone that the address is being validated is done. Perhaps the naming of the method `partitionAddress` can be improved?